lib/elixir/lib/keyword.ex

```diff
@@ -867,6 +867,40 @@ defmodule Keyword do
     raise KeyError, key: key, term: original
   end
 
+  @doc """
+  Replaces the value under `key` using the given function only if
+  `key` already exists in `keywords`.
+
+  In comparison to `replace/3`, this can be useful when it's expensive to calculate the value.
+
+  If `key` does not exist, the original keyword list is returned unchanged.
+
+  ## Examples
+
+      iex> Keyword.replace_lazy([{:a, 1}, {:b, 2}], :a, fn v -> v * 4 end)
+      [{:a, 4}, {:b, 2}]
+
+      iex> Keyword.replace_lazy([{:a, 1}, {:b, 2}], :c, fn v -> v * 4 end)
+      [{:a, 1}, {:b, 2}]
+
+  """
+  @doc since: "1.14.0"
+  @spec replace_lazy(t, key, (existing_value :: value -> new_value :: value)) :: t
+  def replace_lazy(keywords, key, fun)
+      when is_list(keywords) and is_atom(key) and is_function(fun, 1) do
+    do_replace_lazy(keywords, key, fun)
+  end
+
+  defp do_replace_lazy([{key, value} | keywords], key, fun) do
+    [{key, fun.(value)} | delete(keywords, key)]
+  end
+
+  defp do_replace_lazy([{_, _} = e | keywords], key, fun) do
+    [e | replace_lazy(keywords, key, fun)]
```

Comment:
Why is this calling `replace_lazy` instead of `do_replace_lazy`? 
The only way to get to `do_replace_lazy` is from `replace_lazy`. So if you're already in `do_replace_lazy`, then the recursive call should be to `do_replace_lazy`. No need to call `replace_lazy` as it's less efficient.

Reply:
Correct. Please do send a PR!

Reply:
Copy & paste error